crates/bevy_ecs/src/event/base.rs

```diff
@@ -91,9 +92,12 @@ use core::{
 #[diagnostic::on_unimplemented(
     message = "`{Self}` is not an `Event`",
     label = "invalid `Event`",
-    note = "consider annotating `{Self}` with `#[derive(Event)]`"
+    note = "consider annotating `{Self}` with `#[derive(BroadcastEvent)]` or `#[derive(EntityEvent)]`"
 )]
 pub trait Event: Send + Sync + 'static {
+    #[doc(hidden)]
+    type Kind;
```

Comment:
Why an associated type over an associated constant?

Why do we need an associated type or constant at all?

These are genuine questions, and something that we should document here.